kahuna/public/js/services/api/leases.js

```diff
@@ -112,13 +112,13 @@ leaseService.factory('leaseService', [
       const imagesArray = images.toArray ? images.toArray() : images;
       return $q.all(imagesArray.map(image => {
         return image.get().then(apiImage => {
-          const apiLeases = imageAccessor.readLeases(apiImage);
-          const leases = imageAccessor.readLeases(image);
-          const currentLastModified = moment(apiLeases.lastModified);
-          const previousLastModified = moment(leases.lastModified);
-          if (currentLastModified.isAfter(previousLastModified)) {
-            return { image: apiImage, leases: apiLeases };
+          const apiImageAndApiLeases = getApiImageAndApiLeasesIfUpdated(image, apiImage);
```

Comment:
This is now much clearer behind a function call, excellent 👍

Reply:
thanks :)